src/main/scala/project/problem/VrpProblem.scala

```diff
@@ -0,0 +1,79 @@
+package project.problem
+import project.graph.{Edge, Node}
+
+class VrpProblem(val capacity: Int, val depot: Node, nodes: List[Node], val matrix: Map[Edge, Double], val demands: Map[Node, Int])
+    extends BaseProblem(
+      nodes,
+      edges = matrix.keys.toList,
+      startingNode = depot,
+      1
+    ) {
+  var nOTrucks = 0
```

Comment:
This can't be tracked here since `VrpProblem` is not related to a single solution. Multiple ants building different solutions are calling `getPossibleMoves`